kmodloader.c

```diff
@@ -926,32 +939,32 @@
 
 	if (init_module_folders()) {
 		fprintf(stderr, "Failed to find the folder holding the modules\n");
-		ret = -1;
 		goto err;
 	}
 
-	if (get_module_path(argv[1])) {
-		name = argv[1];
-	} else if (!get_module_path(name)) {
+	if (!(path = get_module_path(argv[1])) ||
+	     (path = get_module_path(name))) {
 		fprintf(stderr, "Failed to find %s. Maybe it is a built in module ?\n", name);
-		ret = -1;
 		goto err;
 	}
 
-	ret = insert_module(get_module_path(name), options);
+	ret = insert_module(path, opts);
```

Comment:
I wonder if we can implement some kind of validation to this? Is it sensible to deny something like ../ (and other kind of problematic path) ?

Reply:
Hmmm. One warning gone but it still doesn't like the treatment of `LD_LIBRARY_PATH` env variable, even though it seems that `init_module_folders()` does ultimately test the constructed paths. My impression is that CodeQL can get confused by reusing/aliasing generic `char*` and with tracing data flows. I bet it would be happy if we did `stat()` on `LD_LIBRARY_PATH` components up front...

Need to think about the original code a bit as it's not the clearest thing. And while not related to my PR changes, I'm guessing any future PRs will also trigger this warning unless dealt with somehow.

BTW @Ansuel can I easily run the same CodeQL checks in my own fork?

Reply:
@guidosarducci yes you can, just fork
CodeQL is limited to master branch. They will appear in the security tab on your fork...

OR you can just make a pr on your fork merging to your master branch

We can flag this as false positive if we don't find a not-so-complex solution, in theory it won't be retriggered as it does detect the report already exist and was flagged as false positive (unless the code is changed again in later commits)

Reply:
@Ansuel OK, thanks Christian. Forking to generate our own private CodeQL works and seems really useful. The problem is that it flagged 28 issues for me, with several "Critical". A quick look shows some are real; some are false positives; some are in-between where a bit of code cleanup would help readability for humans and analysis by CodeQL, and might even close an issue as I did above.

These should all be properly reviewed/triaged and addressed as needed. However, they're not directly related to this PR, so I'd suggest doing that separately after first merging this one. Can we keep the one current CodeQL failure without marking as false, just for reference/marking later during a larger review?

Reply:
@Ansuel Are you OK with marking as false and merging?